src/ape/managers/chain.py

```diff
@@ -420,16 +432,34 @@ def __setitem__(self, address: AddressType, contract_type: ContractType):
             contract_type (ContractType): The contract's type.
         """
 
-        if self.get(address) and self._is_live_network:
-            return
+        # Do we need this?
+        # if self.get(address) and self._is_live_network:
```

Comment:
I have no idea why we need this line... If we need it, I'll have to make some sort of "don't make the proxy call" or "don't check the network" flag for testing. Let me know what you think.